js-tests/test-utils.js

```diff
@@ -22,20 +31,50 @@ export function createProvideObject() {
     assetUrls: {},
     dataService: {
       getReports() {
-        return Promise.resolve([
-          { reportId: 1, 'title': 'Report 1' },
-          { reportId: 2, 'title': 'Report 2' },
-          { reportId: 3, 'title': 'Report 3' }
-        ]);
+        return Promise.resolve(mockReports);
+      },
+
+      fetchReportSummary() {
+        return Promise.resolve([]);
       },
 
       saveReportSummary(reportSummary) {
         return Promise.resolve([reportSummary]);
       },
 
+      saveReportSummaries(reportSummaries) {
+        return Promise.resolve([]);
+      },
+
       getReportFields() {
         return Promise.resolve(mockReportFields);
       }
     }
   };
 }
+
```

Comment:
I'm rather proud of this utility. It makes waiting for promise chains to flush and the DOM to update easy and predictable.

Reply:
Seems like a good approach for this type of utility work.  Nice.